src/tests/profiler/native/nongcheap/nongcheap.cpp

```diff
@@ -77,28 +81,84 @@ HRESULT NonGcHeapProfiler::GarbageCollectionFinished()
     else
     {
         // Save very first object ID to compare with EnumerateNonGCObjects
-        firstObj = segments[0].rangeStart;
+        firstObj = frozen_segments[0].rangeStart;
 
         printf("\nGetNonGCHeapBounds (segCount = %u):\n", segCount);
         for (ULONG i = 0; i < segCount; i++)
         {
             printf("\tseg#%u, rangeStart=%p, rangeLength=%u, rangeLengthReserved=%u\n",
-                i, (void*)segments[i].rangeStart, (ULONG)segments[i].rangeLength, (ULONG)segments[i].rangeLengthReserved);
+                i, (void*)frozen_segments[i].rangeStart, (ULONG)frozen_segments[i].rangeLength, (ULONG)frozen_segments[i].rangeLengthReserved);
 
-            if ((ULONG)segments[i].rangeLength > (ULONG)segments[i].rangeLengthReserved)
+            if ((ULONG)frozen_segments[i].rangeLength > (ULONG)frozen_segments[i].rangeLengthReserved)
             {
                 printf("GetNonGCHeapBounds: rangeLength > rangeLengthReserved");
                 _failures++;
             }
 
-            if (!segments[i].rangeStart)
+            if (!frozen_segments[i].rangeStart)
             {
                 printf("GetNonGCHeapBounds: rangeStart is null");
                 _failures++;
             }
+            segment_starts.emplace_back((ULONG)frozen_segments[i].rangeStart);
+            segment_ends.emplace_back((ULONG)frozen_segments[i].rangeStart + (ULONG)frozen_segments[i].rangeLengthReserved);
         }
         printf("\n");
     }
+    hr = pCorProfilerInfo->GetGenerationBounds(MAX_NON_GC_HEAP_SEGMENTS, &segCount, normal_segments);
```

Comment:
I'm guessing we already have other tests that validate GetGenerationBounds() in isolation (@davmason probably knows better exactly what is tested). It seems like the valuable new invariant to test for is confirming that GetGenerationBounds() and GetNonGCHeapBounds() aren't both enumerating the same region of memory.